config/config.go

```diff
@@ -78,6 +87,49 @@ func (config *Config) Validate(path string) ([]string, error) {
 	return deps, nil
 }
 
+// ValidateCloudStoryEnabled creates the list of implicit dependencies.
+func (config *Config) ValidateCloudStoryEnabled(path string) ([]string, error) {
+	cameraName := ""
+	if config.IMUIntegrationEnabled {
+		var ok bool
+		cameraName, ok = config.Camera["name"]
+		if !ok {
+			return nil, utils.NewConfigValidationError(path, errCameraMustHaveName)
+		}
+		dataFreqHz, ok := config.Camera["data_frequency_hz"]
+		if ok {
+			dataFreqHz, err := strconv.Atoi(dataFreqHz)
+			if err != nil {
+				return nil, errors.New("camera[data_frequency_hz] must only contain digits")
+			}
+			if dataFreqHz < 0 {
+				return nil, errors.New("cannot specify camera[data_frequency_hz] less than zero")
+			}
+		}
+	} else {
+		if config.Sensors == nil || len(config.Sensors) < 1 {
+			return nil, utils.NewConfigValidationError(path, errSensorsMustNotBeEmpty)
+		}
+		cameraName = config.Sensors[0]
+
+		if config.DataRateMsec < 0 {
+			return nil, errors.New("cannot specify data_rate_msec less than zero")
+		}
+	}
+
+	if config.ConfigParams["mode"] == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "config_params[mode]")
+	}
+
+	if config.ExistingMap == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "existing_map")
+	}
+
+	deps := []string{cameraName}
+
+	return deps, nil
+}
+
 // GetOptionalParameters sets any unset optional config parameters to the values passed to this function,
 // and returns them.
 func GetOptionalParameters(config *Config, defaultLidarDataRateMsec, defaultIMUDataRateMsec, defaultMapRateSec int, logger golog.Logger,
```

Comment:
This will need to be changed as well, but i am not sure if it is worth changing until more of the cartographer changed are made for cloud story. Thoughts?

Reply:
fine with leaving this as is until the relevant tickets are implemented